src/components/base/outline-alert/outline-alert.ts

```diff
@@ -0,0 +1,64 @@
+import { html, TemplateResult, CSSResultGroup } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import componentStyles from './outline-alert.css.lit';
+import { OutlineElement } from '../outline-element/outline-element';
+
+export type AlertSize = 'small' | 'large';
+export const alertSizes: AlertSize[] = ['small', 'large'];
+
+export type AlertStatusType = 'information' | 'warning' | 'error' | 'success';
+export const alertStatusTypes: AlertStatusType[] = [
+  'information',
+  'warning',
+  'error',
+  'success',
+];
+
+/**
+ * The Outline Alert component
+ *
+ * @element outline-alert
+ * @slot default - The alert contents
+ * @slot outline-alert--header - The header in the alert
+ */
+@customElement('outline-alert')
+export class OutlineAlert extends OutlineElement {
+  static styles: CSSResultGroup = [componentStyles];
+
+  render(): TemplateResult {
+    // The `body` wrapper is used to avoid styles (like border) that are preventing us from styling `:host`.
+    return html`
+      <div id="body" role="${this.isInteractive ? 'alertdialog' : 'alert'}">
+        ${this.shouldShowIcon === true
+          ? html`
+              <div id="icon">
+                <!--@todo include icon when we have that ready.-->
+              </div>
+            `
+          : html``}
+        ${this.size === 'large'
+          ? html`
+              <div id="header">
+                <slot name="outline-alert--header">${this.statusType}</slot>
+              </div>
+            `
+          : html``}
+        <div id="message">
+          <slot></slot>
+        </div>
+      </div>
+    `;
+  }
+
+  @property({ type: Boolean })
```

Comment:
The `@property`'s should be moved above the `render()`. Also can we add doc blocks for what these props control.

Reply:
Thanks, @madhaze!

I appreciate the review and your perspective. It's incredibly helpful, because while I'm trying to create clear code, I don't have the perspective of someone else. :) I can only make guesses.

I've looked at the Lit examples (https://lit.dev/docs/) and see that the expectation is to put the properties first. I think that expectation makes sense and I got a little carried away with putting the `render()` method first.

I've also been trying to limit comments when the purpose of variables / properties / methods is clear, but I think the `why` of `isInteractive` is not so clear. What do you think of the others like `size`, `statusType`, `shouldShowIcon`?

Thanks again! Always appreciate the feedback.